### PR TITLE
Fix `build_requestinfo` in OpenSSL1.1.0+ environments

### PR DESCRIPTION
**Pull request by stanislavlevin**
Since OpenSSL 1.1.0 the `req_info` field of X509_REQ structure is
no longer a pointer to X509_REQ_INFO. This results in a crash of
`build_requestinfo` in environments having OpenSSL1.1.0+ (libcrypto).

With this patch, the X509_REQ definition becomes the version dependent.
Both OpenSSL1.0.x and OpenSSL1.1.x are supported.

Fixes: https://pagure.io/freeipa/issue/7937

### CODE REVIEW

File: ipaclient/csrgen_ffi.py [hunk 1]
```diff
@@ -118,10 +149,7 @@
 char *ERR_error_string(unsigned long e, char *buf);
 ''')  # noqa: E501
 
-_libcrypto = _ffi.dlopen(ctypes.util.find_library('crypto'))
```
Comment (abbra) [inline ipaclient/csrgen_ffi.py]: Why are you removing this line? The code will break because we reference `_libcrypto` below.

Comment (stanislavlevin) [reply ipaclient/csrgen_ffi.py]: @abbra, thank you for looking at this. 
Actually, we need to know OpenSSL version ( `OpenSSL_version_num` / `SSLeay`) before parsing the related C code (`cdef`). Thus, the loading libcrypto in ABI mode happened earlier (see line 13, please).

Comment (abbra) [reply ipaclient/csrgen_ffi.py]: Ah, thanks. I missed that part. It looks OK to me but fedora-29/caless failed due to a timeout. I restarted the test.

**Comment by abbra** [comment]
Tests completed. @tiran, please help with the review too.

**Comment by abbra** [comment]
I had to pull this commit into my tree https://github.com/abbra/freeipa/pull/11 to make it possible to run Tox tests on Fedora 30 (that pull request is against my own fork and is running tests against Azure Pipelines runners). So ACK.

**Comment by fcami** [comment]
master:

* ac6568dcf58ec8d06df5493d14a28aa41845d4ef Fix `build_requestinfo` in OpenSSL1.1.0+ environments

**Comment by stanislavlevin** [comment]
@abbra, thank you!

**Pull request closed without merging**